### PR TITLE
Enable SwiftLint - solves(#23)

### PR DESCRIPTION
**Pull request by adamstener**
This PR enables SwiftLint in the build process and modifies the existing code to comply with all the SwiftLint rules.

https://github.com/zcash/secant-ios-wallet/issues/23

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [x] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [x] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/mater/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage befor and after your changes and when possible, leave it in a better place. [Learn More...](../blob/master/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/master/README.md), [LICENSE.md](../blob/master/LICENSE.md), and [Architecture.md](../blob/master/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -117,7 +117,7 @@ identifier_name:
     - z
 
 indentation_width:
-  indentation_width: 2
+  indentation_width: 4
```
Comment (adamstener) [inline .swiftlint.yml:120]: I wanted to bring this one to your attention @pacu.  I saw we had 2 previously in the SwiftLint file but 4 is Xcodes default.  Let me know if this makes sense or if you feel that we should re work this to go with 2.

Comment (pacu) [reply .swiftlint.yml:120]: that's fine. good catch

File: xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___ScreenViewModel.swift [hunk 2]
```diff
@@ -3,6 +3,4 @@
 import Foundation
 import Combine
```
Comment (pacu) [inline xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___ScreenViewModel.swift:5]: This is a template file so it's fine that it has blank space. It's supposed to be modified

File: xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___Screen.swift [hunk 3]
```diff
@@ -2,16 +2,15 @@
...
 struct ___FILEBASENAMEASIDENTIFIER___: View {
-    @State var router: ___FILEBASENAMEASIDENTIFIER___Router?
-    
     @ObservedObject var viewModel: ___FILEBASENAME___ViewModel
-    
+
+    @State var router: ___FILEBASENAMEASIDENTIFIER___Router?
+
     var body: some View {
-        Text(/*@START_MENU_TOKEN@*/"Hello, World!"/*@END_MENU_TOKEN@*/)
```
Comment (pacu) [inline xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___Screen.swift:14]: This is a template file, it's ok that it has menu token markers

File: xctemplates/Router Templates/Router.xctemplate/___FILEBASENAME___Router.swift [hunk 4]
```diff
@@ -4,28 +4,16 @@ import Foundation
...
-    // MARK: - Internal vars
     var services: Services
-        
-    // MARK: - Initialization
 
     init(services: Services) {
         self.services = services
     }
-    
-    // MARK: - Methods
```
Comment (pacu) [inline xctemplates/Router Templates/Router.xctemplate/___FILEBASENAME___Router.swift:22]: Why did you remove this pragma marks?

**Review by pacu — CHANGES_REQUESTED** [review]

File: xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___Screen.swift (hunk 3 above)
Comment (adamstener) [reply xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___Screen.swift:14]: Yes, it's totally ok to have token markers and I can put them back if you'd like.  I just didn't see the value in making someone accept the token default to have a valid generated file, given that they will likely not be needing a Text view at all and will need to put something else there altogether.

File: xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___ScreenViewModel.swift (hunk 2 above)
Comment (adamstener) [reply xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___ScreenViewModel.swift:5]: While this is a template file, and blank space is ok here because we are not linting these files, this does cause the template to create a file that is invalid.  I updated the template to make sure that every file generated is valid from the beginning.

File: xctemplates/Router Templates/Router.xctemplate/___FILEBASENAME___Router.swift (hunk 4 above)
Comment (adamstener) [reply xctemplates/Router Templates/Router.xctemplate/___FILEBASENAME___Router.swift:22]: Pragma marks are typically used at the file level to help explain sections of your code.  Adding them _within_ a class is also a valid use case but typically only done once those sections are large enough to not fit on screen and are needed to help navigate.  Inner class pragma marks are also a sign that your class may be too large and you could consider breaking it up in a different way to avoid this.  Including pragma marks by default at the class level before knowing if you really need them or considering alternatives just seemed like something we could avoid.

File: xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___Screen.swift (hunk 3 above)
Comment (adamstener) [reply xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___Screen.swift:14]: This is also how the other template that contains a View works.

Comment (pacu) [reply xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___Screen.swift:14]: Text Views are the default placeholders on SwiftUI projects. It's fine.

File: xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___ScreenViewModel.swift (hunk 2 above)
Comment (pacu) [reply xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___ScreenViewModel.swift:5]: I had the same feeling initially, but then realized that a scaffolding template it's supposed to be that: something that must be changed and turned into valid functioning code. It actually follows the red-green-refactor logic, where you add code that knowingly fails, then you make it work then refactor.

File: xctemplates/Router Templates/Router.xctemplate/___FILEBASENAME___Router.swift (hunk 4 above)
Comment (pacu) [reply xctemplates/Router Templates/Router.xctemplate/___FILEBASENAME___Router.swift:22]: You make a good point there.

How can we enforce code ordering without them? Shall we add a code guideline? Comments that are not marks?

Comment (adamstener) [reply xctemplates/Router Templates/Router.xctemplate/___FILEBASENAME___Router.swift:22]: Yeah it will have to be done through code review and we could certainly add structure into guidelines as something that will be looked for.

File: xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___ScreenViewModel.swift (hunk 2 above)
Comment (adamstener) [reply xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___ScreenViewModel.swift:5]: Sounds good I can get on board with that, I'll make the change.

File: xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___Screen.swift (hunk 3 above)
Comment (adamstener) [reply xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___Screen.swift:14]: I can add this token back in, should I also update the other template to use this as well for consistency sake?

Comment (pacu) [reply xctemplates/Router Templates/SwiftUI Screen.xctemplate/___FILEBASENAME___Screen.swift:14]: yes, that would be good

File: xctemplates/Router Templates/Router.xctemplate/___FILEBASENAME___Router.swift (hunk 4 above)
Comment (pacu) [reply xctemplates/Router Templates/Router.xctemplate/___FILEBASENAME___Router.swift:22]: I'll add that

**Review by pacu — CHANGES_REQUESTED** [review]
Let's add those small changes and call it done :)

**Review by pacu — APPROVED** [review]
